pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -226,6 +232,10 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		return fmt.Errorf("a vm %s/%s already exists", config.Folder, spec.Name)
 	}
 
+	if _, err := session.Finder.ResourcePool(ctx, config.ResourcePool); err != nil {
```

Comment:
We should validate that the resource pool exists only when it is specified (i.e. `if config.ResourcePool !=  ""` to be clearer) not to break backward compatibility.

Reply:
okay makes sense and is changed